pkg/tools/generate-doc.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/gluster/glusterd2/pkg/api"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const endpointsTable = `# API Endpoints
+
```

Comment:
TODO: I'll have to learn to nest templates here. I'll do it later. The generated markdown doc as lot of links but they aren't rendered, so it's not a problem.